common/src/main/java/com/microsoft/identity/common/internal/fido/PasskeyFidoChallengeHandler.kt

```diff
@@ -0,0 +1,129 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.fido
+
+import android.webkit.WebView
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import com.microsoft.identity.common.internal.ui.webview.challengehandlers.IChallengeHandler
+import com.microsoft.identity.common.java.constants.FidoConstants
+import com.microsoft.identity.common.java.opentelemetry.AttributeName
+import com.microsoft.identity.common.java.opentelemetry.OTelUtility
+import com.microsoft.identity.common.java.opentelemetry.SpanName
+import com.microsoft.identity.common.logging.Logger
+import io.opentelemetry.api.trace.SpanContext
+import io.opentelemetry.api.trace.StatusCode
+import kotlinx.coroutines.launch
+
+/**
+ * Handles a FidoChallenge by either creating or authenticating with a passkey.
+ */
+class PasskeyFidoChallengeHandler
+/**
+ * Creates a PasskeyFidoChallengeHandler instance.
+ * @param fidoManager IFidoManager instance.
+ * @param webView Current WebView.
+ * @param spanContext Current spanContext, if present.
+ * @param lifecycleOwner instance to get coroutine scope from.
+ */(
+    private val fidoManager: IFidoManager,
+    private val webView: WebView,
+    private val spanContext : SpanContext?,
+    private val lifecycleOwner: LifecycleOwner?
+) : IChallengeHandler<IFidoChallenge, Void> {
+    val TAG = PasskeyFidoChallengeHandler::class.simpleName.toString()
+    val span = if (spanContext != null) {
+        OTelUtility.createSpanFromParent(SpanName.Fido.name, spanContext)
+    } else {
+        OTelUtility.createSpan(SpanName.Fido.name)
+    }
+
+    override fun processChallenge(fidoChallenge: IFidoChallenge): Void? {
+        val methodTag = "$TAG:processChallenge"
+        span.setAttribute(
+            AttributeName.fido_challenge.name,
+            fidoChallenge::class.simpleName.toString());
+        var assertion: String
+        if (lifecycleOwner != null
+            && fidoChallenge is AuthFidoChallenge) {
+            lifecycleOwner.lifecycleScope.launch {
+                try {
+                    assertion = fidoManager.authenticate(fidoChallenge)
+                    span.setStatus(StatusCode.OK)
+                } catch (e: Exception) {
+                    assertion = e.message.toString()
+                    Logger.error(methodTag, assertion, e)
+                    span.setStatus(StatusCode.ERROR)
+                }
+                respondToChallenge(fidoChallenge.submitUrl, assertion, fidoChallenge.context)
+            }
+            return null
+        }
+        var errorMessage = "Failed based on parameter values."
+        if (lifecycleOwner == null) {
+            errorMessage += " Cannot get lifecycle owner needed for FIDO API calls."
+        }
+        if (fidoChallenge !is AuthFidoChallenge) {
```

Comment:
I'm wondering if we need this check.

If this class only supports AuthFidoChallenge, then maybe we should just pass AuthFidoChallenge here in the first place (not IFidoChallenge)

essentially turning this class into IChallengeHandler<AuthFidoChallenge, Void>

Reply:
and if you get something else in the caller layer - you don't even need to create this object in the first place, just fail.

Reply:
I would expect the Registration flow to be in a separate challenge handler.

Reply:
The parameters and method calls are different for reg and auth. Makes sense to have different challenge handlers. 

( for some context: The original design considered both auth and reg, and at the time, we didn't know when CredMan would start supporting security keys. So my initial idea was that:
- challenges would have auth and reg types
- challenge handlers would have a passkey and a securitykey type (each one would handle reg/auth), and the associated factory would figure out which one we need there.
Things have changed, and now we know that CredMan will eventually support both passkeys and security keys and will provide dialogs to choose between passkey and security key itself (for auth at least, not sure about reg) )